### PR TITLE
Feat:  External link component

### PR DESCRIPTION
**Pull request by kmaid**

### CODE REVIEW

**Review by fampinheiro — COMMENTED** [review]
image

The text doesn't seem to be centered vertically.

File: packages/react-components/src/molecules/ExternalLink.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import css from '@emotion/css';
+import { Link } from '../atoms';
+import { externalLinkIcon } from '../icons';
+import { fern, pine } from '../colors';
+import { perRem } from '../pixels';
+
+const containerStyles = css({
+  display: 'flex',
+});
+const styles = css({
```
Comment (jeysal) [inline packages/react-components/src/molecules/ExternalLink.tsx:11]: Can you organise and group these properties a bit? E.g. if `padding` isn't `12 - borderWidth`, nor located close to the border definition, you're not going to remember to adjust the padding if the border width is changed

File: packages/react-components/src/molecules/__tests__/ExternalLink.test.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import ExternalLink from '../ExternalLink';
+
+it('renders an external link', () => {
+  const { getByRole } = render(<ExternalLink href="http://example.com" />);
+  expect(getByRole('link')).toHaveAttribute('href', 'http://example.com');
+});
```
Comment (jeysal) [inline packages/react-components/src/molecules/__tests__/ExternalLink.test.tsx:9]: renders the label
renders an external link icon (which should prompt us to give the icon a proper `<title>`)

**Review by jeysal — CHANGES_REQUESTED** [review]

File: packages/react-components/src/molecules/__tests__/ExternalLink.test.tsx (hunk 2 above)
Comment (jeysal) [reply packages/react-components/src/molecules/__tests__/ExternalLink.test.tsx:12]: :point_up:

File: packages/react-components/src/organisms/__tests__/SharedResearchCard.test.tsx [hunk 3]
```diff
@@ -64,7 +64,7 @@ it('displays link component when link property present', () => {
   const { getByText } = render(
     <SharedResearchCard {...libraryCardProps} link={'https://example.com'} />,
   );
-  const link = getByText(/external\slink/i).closest('a');
+  const link = getByText(/external\slink/i, { selector: 'div' }).closest('a');
```
Comment (jeysal) [inline packages/react-components/src/organisms/__tests__/SharedResearchCard.test.tsx:67]:
```suggestion
  const link = getByText(/external\slink/i, { selector: 'a *' }).closest('a');
```
I think, if I understood correctly what's going on

**Review by jeysal — APPROVED** [review]

**Pull request merged**